### PR TITLE
[AMD] rocm 7.2 image release, PR test, Nightly Test

### PR DESCRIPTION
**Pull request by yctseng0211**
## Motivation

Enable Rocm 7.2 

## Modifications

ROCM 7.2
- PR Test
- Nightly Test
- cleanup vllm dependencies

## Accuracy Tests

Nightly All green : https://github.com/sgl-project/sglang/actions/runs/21875099676

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by michaelzhang-ai — APPROVED** [review]

File: .github/workflows/release-docker-amd-rocm720-nightly-preview.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: Release Docker Images ROCm 7.2.0 Preview (AMD)
+on:
```
Comment (michaelzhang-ai) [inline .github/workflows/release-docker-amd-rocm720-nightly-preview.yml:2]: todo: add cron schedule

File: scripts/ci/amd/amd_ci_start_container.sh [hunk 2]
```diff
@@ -136,16 +155,56 @@ find_latest_image() {
   echo "Error: no ${gpu_arch} image found in the last 7 days for base ${base_tag}" >&2
   echo "Using hard-coded fallback…" >&2
   if [[ "${gpu_arch}" == "mi35x" ]]; then
-    echo "rocm/sgl-dev:v0.5.5-rocm700-mi35x-20251110"
+    echo "rocm/sgl-dev:v0.5.8.post1-rocm720-mi35x-20260211-preview"
```
Comment (michaelzhang-ai) [inline scripts/ci/amd/amd_ci_start_container.sh]: keep rocm 700 fallback image as well? (rocm-version option)

File: python/sglang/srt/layers/quantization/fp8_kernel.py [hunk 3]
```diff
@@ -1537,6 +1541,37 @@ def per_token_group_quant_mla_deep_gemm_masked_fp8(
...
 if _is_hip:
 
+    def _native_dynamic_per_token_quant_fp8(output, input, scale):
+        """Native PyTorch fallback for dynamic per-token FP8 quantization when vLLM is unavailable."""
+        M, N = input.shape
+        eps = 1e-12
+        # Compute per-token scale
+        absmax = input.abs().max(dim=1, keepdim=True).values
+        absmax = torch.clamp(absmax, min=eps)
+        scale_val = absmax / fp8_max
```
Comment (yichiche) [inline python/sglang/srt/layers/quantization/fp8_kernel.py]: fp8_min / fp8_max  / fp8_dtype is defined as global variable in his script, should be fine. 

Ref:
if is_fp8_fnuz():
    fp8_dtype = torch.float8_e4m3fnuz
    fp8_max = 224.0
else:
    fp8_dtype = torch.float8_e4m3fn
    fp8_max = torch.finfo(fp8_dtype).max
fp8_min = -fp8_max

**Review by HaiShaw — APPROVED** [review]
Changes are AMD (_is_hip or _is_aiter) constrained.

**Review by HaiShaw — COMMENTED** [review]
rename `release-docker-amd-rocm720-nightly-preview.yml` to `release-docker-amd-rocm720-nightly.yml` next time; consider to add `release-docker-amd-rocm720.yml` next time.
